### DISCUSSION TITLE
Generate an .img and after use PiShrink

### DISCUSSION THREAD

**Discussion by PabloPittana** [Drewsif/PiShrink#279] [Ideas]
I'm new making backups for my RPI4 and I'm following a guide to make an .img and after run your script to reduce it. And I think that you're able to make your script to be same result with one command.

Now I'm running this command to concatenate two commands:
sudo dd if=/dev/mmcblk0 of=/your/route/light_backup_$(date '+%Y_%m_%d').img bs=1M && sudo pishrink.sh /your/route/light_backup_$(date '+%Y_%m_%d').img

The first command make me a huge .img and the second reduce that .img. Think about to only run one command with pishrink to be able to do the same thing.
Some thing like this:
sudo pishrink.sh -A -R /the/route/backup.img
[-A to distinge you want to make the backup and reduce after] and [-R to put your destination route with the name of .img]

Thank you for you're time!! :D

**Comment by DonatelloLV** [comment]
Hi @PabloPittana,

I'm having the same backup need as you --> https://lemmy.ml/post/22987661

here the command that I sue to generate the backup of the boot SDcard

```bash
dd if=/dev/sdb of=raspberry_pi_backup.img.gz conv=sparse,notrunc bs=4096 status=progress
```
This is generating a backup of the same size of the sdcard ! (because it backup unused space too, so the full partition)
I've read that `conv=sparse` should not write unused space, but apparently it's not the case..
I've found `status=progress` useful too.

**Comment by dreidel848** [comment]
Yes, it would be great to get a shrunk image in one go. 
Anyway thanks for you effort.  Useful stuff.
